File: Crazy8s/Crazy8s/Crazy8ProvidedCode/JavaCrazy8/Game (2016_10_11 04_07_57 UTC).java
/**
 * Play the game Crazy Eights.
 * The computer player in this version always draws a card.
 * If the deck runs out, the program will throw an exception and terminate.
 */
public class Game {
  private Deck deck;
  private Pile pile;
  private HumanPlayer human;
  private ComputerPlayer computer;
  
  /**
   * Initialize game by creating and shuffling the deck,
   * dealing one card (other than an 8) to the discard pile,
   * and dealing 7 cards to each player.
   */
  public Game() {
    deck = new Deck();
    do {
      deck.shuffle();
    } while (deck.isTopCardAnEight());
    pile = new Pile();
    pile.acceptACard(deck.dealACard());
    
    human = new HumanPlayer(deck);
    computer = new ComputerPlayer(deck);
    
  }
  
  /**
   * Play one complete game.
   */
  public void play() { 
    
    do {
      human.takeATurn(deck, pile);
      if (!human.isHandEmpty()) {
        computer.takeATurn(deck, pile);
      }
    } while (!(human.isHandEmpty() || computer.isHandEmpty()));
    if (human.isHandEmpty()) {
      System.out.println("Congratulations!");
    }
    else {
      System.out.println("Thanks for being a good loser.");
    }
  }
  
  public static void main(String[] args) {
    Game game = new Game();
    game.play();
  }
}

File: Crazy8s/Crazy8s/Game (2016_10_11 04_07_57 UTC).js
"use strict";

function Game(){
    this.deck = new Deck();
    do {
	this.deck.shuffle();
    }
    while(this.deck.isTopCardAnEight());
    this.pile = new Pile();
    this.pile.acceptACard(this.deck.dealACard());
    
    this.human = new HumanPlayer(this.deck);
    this.computer = new ComputerPlayer(this.deck);
}

Game.prototype = {
    play: function(){
	do{
	    this.human.takeATurn(this.deck, this.pile);
	    if(!(this.human.isHandEmpty())){
		this.computer.takeATurn(this.deck, this.pile);
	    } 
	}
	while(!(this.human.isHandEmpty() || this.computer.isHandEmpty()));
	if(this.human.isHandEmpty){
	    alert("Congratulations!");
	}
	else{
	    alert("Thanks for being a good loser.");
	}
    }
};
